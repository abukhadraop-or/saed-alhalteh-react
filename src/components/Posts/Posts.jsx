import MetaInfo from 'components/MetaInfo/MetaInfo';
import PropTypes from 'prop-types';
import React from 'react';
import {
  Container,
  Like,
  PostContent,
  ReadMoreDiv,
  ReadMoreLink,
  UserDataContainer,
} from './posts.styles';

/**
 * Posts details.
 *
 * @param {object} props The props object.
 * @param {object[]} props.posts Posts.
 *
 * @return {JSX.Element} Posts component.
 */
function Posts({ posts }) {
  return posts.length === 0 ? (
    <Container>
      <p> No articles are here... yet. </p>
    </Container>
  ) : (
    posts.map((post) => (
      <Container key={post.slug}>
        <UserDataContainer>
          <MetaInfo author={post.author} createdAt={post.createdAt} />
          <Like count={post.favoritesCount} />
        </UserDataContainer>
        <PostContent to={{ pathname: '/posts', state: { detail: post } }}>
          <h2>{post.title}</h2>
          <p>{post.description}</p>
        </PostContent>
        <ReadMoreDiv>
          <ReadMoreLink to={{ pathname: '/posts', state: { detail: post } }}>
            Read more...
          </ReadMoreLink>
        </ReadMoreDiv>
      </Container>
    ))
  );
}

Posts.defaultProps = {
  posts: [],
};

Posts.propTypes = {
  posts: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      description: PropTypes.string,
      favoritesCount: PropTypes.number,
      createdAt: PropTypes.string,
      author: PropTypes.shape({
        username: PropTypes.string,
        bio: PropTypes.string,
        image: PropTypes.string,
        following: PropTypes.bool,
      }),
    })
  ),
};

export default Posts;
