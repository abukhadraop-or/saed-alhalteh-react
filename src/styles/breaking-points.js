const breakingPoints = {
  mobile: '30rem',
  tablets: '48rem',
  smallScreen: '64.0625rem',
  largeScreen: '75rem',
};

export default breakingPoints;
