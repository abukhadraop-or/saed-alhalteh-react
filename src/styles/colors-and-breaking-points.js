export const colors = {
  black: '#000',
  blueViolet: '#4F52B0',
  concrete: '#f3f3f3',
  dustyGrey: '#999',
  edward: '#acb9b4',
  fern: '#5cb85c',
  goblin: '#347634',
  gray: '#868686',
  mineShaft: '#333333',
  oBlack: 'rgba(0,0,0,0.75)',
  o3Black: 'rgba(0,0,0,0.3)',
  oFern: 'rgba(92,184,92,0.5)',
  silver: '#c8c8c8',
  white: '#fff',
};

export const breakingPoints = {
  mobile: '480px',
  tablets: '768px',
  smallScreen: '1025px',
  largeScreen: '1200px',
};
