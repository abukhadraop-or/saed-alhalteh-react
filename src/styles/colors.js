const colors = {
  black: '#000000',
  blueViolet: '#4f52b0',
  concrete: '#f3f3f3',
  darkGreen: '#347634',
  dustyGrey: '#999999',
  gray: '#868686',
  green: '#5cb85c',
  lightGrey: '#acb9b4',
  mineShaft: '#333333',
  partiallyDarkBlack: 'rgba(0,0,0,0.75)',
  silver: '#c8c8c8',
  transparentBlack: 'rgba(0,0,0,0.3)',
  transparentGreen: 'rgba(92,184,92,0.5)',
  white: '#ffffff',
};

export default colors;
