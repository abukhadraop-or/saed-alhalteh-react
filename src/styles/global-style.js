import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  body {
    font-family:'Titillium Web', sans-serif;
    margin: 0;
  }
`;

export default GlobalStyle;
