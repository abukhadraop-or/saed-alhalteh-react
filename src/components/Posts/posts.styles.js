import LikeComponent from 'components/Like';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import breakingPoints from 'styles/breaking-points';
import colors from 'styles/colors';

export const Container = styled.div`
  padding: 0 0.625rem;

  @media (min-width: ${breakingPoints.mobile}) {
    margin: 0 auto;
  }

  @media (min-width: ${breakingPoints.tablets}) {
    margin: 0 0 0 2.5rem;
    padding: 0 2rem 0 0.625rem;
  }

  @media (min-width: ${breakingPoints.largeScreen}) {
    margin: 0 0 0 6.25rem;
  }
`;

export const UserDataContainer = styled.div`
  display: flex;
  padding: 0 1.25rem 0 0;
`;

export const Like = styled(LikeComponent)`
  & span {
    margin: 0;
  }

  @media (min-width: ${breakingPoints.tablets}) {
    margin: 0 0.9375rem 0 auto;
  }
`;

export const PostContent = styled(Link)`
  display: inline-block;
  text-decoration: none;

  & h2 {
    color: ${colors.black};
    font-size: 1.5rem;
    font-weight: 600;
    margin: 0.5rem 0;
  }

  & p {
    color: ${colors.dustyGrey};
    font-size: 1rem;
    font-weight: 300;
    margin: 0.25rem 0;
  }
`;

export const ReadMoreDiv = styled.div`
  padding: 0.5rem 0 1.5rem;
`;

export const ReadMoreLink = styled(Link)`
  color: ${colors.dustyGrey};
  cursor: pointer;
  font-size: 0.75rem;

  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`;
