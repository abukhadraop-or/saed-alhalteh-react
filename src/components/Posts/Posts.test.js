import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Posts from '.';
import { PostContent } from './posts.styles';

Enzyme.configure({ adapter: new Adapter() });

describe('<Posts /> tests', () => {
  it('matches the snapshot when there is no posts', () => {
    const wrapper = shallow(<Posts />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('hides "No articles are here" and display posts details, when there is posts', () => {
    const posts = [
      {
        id: 'gvnzew4eh6d87v3',
        slug: 'how-to-train-your-dragon',
        title: 'How to train your dragon',
        description: 'Ever wonder how?',
        body: '1 Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
        tagList: ['dragons', 'training', 'odd', 'only the first Article'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 27,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image:
            'https://upload.wikimedia.org/wikipedia/en/thumb/2/24/Crystal_personal.svg/1024px-Crystal_personal.svg.png',
          following: false,
        },
      },
      {
        slug: 'how-to-train-your-dragon-2',
        title: 'How to train your dragon 2',
        description: 'So toothless',
        body: '2 Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
        tagList: ['dragons', 'training', 'even'],
        createdAt: '2016-02-18T03:22:56.637Z',
        updatedAt: '2016-02-18T03:48:35.824Z',
        favorited: false,
        favoritesCount: 0,
        author: {
          username: 'jake',
          bio: 'I work at statefarm',
          image: 'https://i.stack.imgur.com/xHWG8.jpg',
          following: false,
        },
      },
    ];
    const wrapper = shallow(<Posts posts={posts} />);

    expect(
      wrapper
        .find('p')
        .at(0)
        .matchesElement(<p> No articles are here... yet. </p>)
    ).toBeFalsy();
    expect(wrapper.find(PostContent).length).toBe(2);
  });
});
