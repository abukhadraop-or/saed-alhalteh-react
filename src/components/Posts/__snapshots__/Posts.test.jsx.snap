// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> tests matches the snapshot when there is no posts 1`] = `
<styled.div>
  <p>
     No articles are here... yet. 
  </p>
</styled.div>
`;
